company-template.el

```diff
@@ -65,15 +66,27 @@
            (> pos (point))
            (< pos minimum)
            (setq minimum pos)))
-    (push-mark)
-    (goto-char minimum)
-    (company-template-remove-field (company-template-field-at start))))
+    (let ((start-ovl (company-template-field-at start)))
+      (if delete-template
```

Comment:
Sure, this was just a quick way to have something to discuss based on the feature request in #657 .